# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/nsSocketTransportService2.h
# Commit: 3e4914be9f30
# Full Hash: 3e4914be9f30f747091ed2a4bccbd8a56e40a0e6
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2023-05-26 21:54:33
# Regressor Bug: 1818998
# File Overlap Count: 1
# Description:
#   Bug 1834640 - Part 2: Properly shutdown nsSocketTransportService during xpcom-shutdown-threads, r=jesup,necko-reviewers
#   
#   This also cleans up some of the unnecessary flags & memebrs which were held
#   during the changes in bug 1818998. mSelf was unncessary as the runnable
#   reference will be keeping the instance alive for the same lifetime, and
# ==============================================================================

diff -r f762e1cce59c -r 3e4914be9f30 netwerk/base/nsSocketTransportService2.h
--- a/netwerk/base/nsSocketTransportService2.h	Fri May 26 16:50:22 2023 +0000
+++ b/netwerk/base/nsSocketTransportService2.h	Fri May 26 16:50:22 2023 +0000
@@ -160,9 +160,6 @@
   Atomic<bool> mInitialized{false};
   // indicates whether we are currently in the process of shutting down
   Atomic<bool> mShuttingDown{false};
-  Atomic<bool> mSocketThreadShutDown{false};
-  // Effectively owned by the SocketThread
-  RefPtr<nsSocketTransportService> mSelf;
 
   Mutex mLock{"nsSocketTransportService::mLock"};
   // Variables in the next section protected by mLock
